src/extension/mapml/src/main/java/org/geoserver/mapml/tcrs/TiledCRS.java

```diff
@@ -386,8 +396,52 @@ public List<TileCoordinates> getTilesForExtent(LatLngBounds extent, int zoom, lo
         }
         return getTilesForExtent(pb, zoom, start);
     }
+
     // extent is in projected but not scaled units (e.g. meters)
 
+    /**
+     * Get the Min zoom level that matches the given denominator
+     *
+     * @param denominator the denominator to match
+     * @return the zoom level that matches the given denominator
+     */
+    public Integer getMinZoomForDenominator(double denominator) {
+        // handles the case where denominator is larger than the largest scale or is infinity
+        if (denominator == Double.POSITIVE_INFINITY
+                || denominator >= convertGroundResolutionToScale(scales[0])) return 0;
```

Comment:
> @prushforth can we clarify the code? We can either stop calling them scales, and calling them resolutions instead (and avoid the "1/res" math, just use the resolution directly instead)?

yes, it would be better to use resolutions directly, but it may involve breakage / refactoring

> Checking actual use, it seems the "scales" are not used 

good, hopefully that makes it straightforward tbd.

The actual numbers are duplicated, which is not good, in the MapMLGridSets.java class, where they are used in resolutions.

Sorry for the confusion.